### PR TITLE
Add -v/--version flag

### LINKED ISSUE #210: Add --version option
Now that people are relying on this to produce shapes for production, they need to be able to check which version of the thriftrw-go binary they have. Additionally, we should probably put this version at the top of generated code in comments.

### PR DESCRIPTION
**Pull request by abhinav**
$ thriftrw-go --version
    thriftrw v0.2.0

`make test` verifies that the version number on the top of the changelog
matches `--version`. Example failure:

    Version number in version.go does not match CHANGELOG.md
    Want: thriftrw v0.2.0
     Got: thriftrw v0.2.1
    make: *** [verifyVersion] Error 1

Resolves #210

CC @prashantv @kriskowal @breerly @bombela

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,9 @@ Releases
 v0.2.0 (unreleased)
 -------------------
 
--   Add a plugin system.
+-   Added a `-v`/`--version` flag.
+
```
Comment (breerly) [inline CHANGELOG.md]: No spaces here

File: version.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+var version = "v0.2.0"
```
Comment (breerly) [inline version.go]: any reason this shouldn't be exported?

File: Makefile [hunk 3]
```diff
@@ -59,8 +61,19 @@ else
...
+		echo "Want: thriftrw $(WANT_VERSION)"; \
+		echo " Got: $$(./thriftrw-go --version)"; \
+		exit 1; \
+	else \
+		echo "thriftrw $(WANT_VERSION)"; \
+	fi
+
 .PHONY: test
-test: build
+test: build verifyVersion
```
Comment (breerly) [inline Makefile]: nice

**Comment by breerly** [comment]
lgtm

File: version.go (hunk 2 above)
Comment (abhinav) [inline version.go]: You shouldn't import the `main` package directly, and even if you do, it currently doesn't export anything else either.

**Pull request merged**